### PR TITLE
Move `UnregisterWSL()` to `wsl` package

### PR DESCRIPTION
**Pull request by adamkpickering**
It makes more sense to me to have all of the WSL functionality under the `wsl` package. I know we didn't discuss this as a team at all, so feel free to close if you disagree.

### CODE REVIEW

File: src/go/rdctl/pkg/factoryreset/factory_reset_windows.go [hunk 1]
```diff
@@ -28,15 +28,15 @@ import (
...
-	"syscall"
 	"unsafe"
 
 	"github.com/rancher-sandbox/rancher-desktop/src/go/rdctl/pkg/directories"
 	"github.com/sirupsen/logrus"
 	"golang.org/x/sys/windows"
-	"golang.org/x/text/encoding/unicode"
 )
 
+const CREATE_NO_WINDOW = 0x08000000
```
Comment (mook-as) [inline src/go/rdctl/pkg/factoryreset/factory_reset_windows.go:38]: Is this used? (The only `CREATE_NO_WINDOW` in this file refers to the export from `windows`, which makes sense…)

File: src/go/rdctl/pkg/wsl/wsl_windows.go [hunk 2]
```diff
@@ -21,7 +25,34 @@ type WSL interface {
...
+		if s == "rancher-desktop" || s == "rancher-desktop-data" {
+			wslsToKill = append(wslsToKill, s)
+		}
+	}
+
+	for _, wsl := range wslsToKill {
+		cmd := exec.Command("wsl", "--unregister", wsl)
+		cmd.SysProcAttr = &windows.SysProcAttr{CreationFlags: windows.CREATE_NO_WINDOW}
+		if err := cmd.Run(); err != nil {
+			logrus.Errorf("Error unregistering WSL %s: %s\n", wsl, err)
```
Comment (mook-as) [inline src/go/rdctl/pkg/wsl/wsl_windows.go:52]:
```suggestion
			logrus.Errorf("Error unregistering WSL distribution %s: %s\n", wsl, err)
```

File: src/go/rdctl/pkg/wsl/wsl_windows.go [hunk 3]
```diff
@@ -21,7 +25,34 @@ type WSL interface {
...
+	if err != nil {
+		return fmt.Errorf("error getting current WSLs: %w", err)
+	}
+	decoder := unicode.UTF16(unicode.LittleEndian, unicode.IgnoreBOM).NewDecoder()
+	actualOutput, err := decoder.String(string(rawBytes))
+	if err != nil {
+		return fmt.Errorf("error getting current WSLs: %w", err)
+	}
+	actualOutput = strings.ReplaceAll(actualOutput, "\r", "")
+	wsls := strings.Split(actualOutput, "\n")
```
Comment (mook-as) [inline src/go/rdctl/pkg/wsl/wsl_windows.go:40]: Consider using [`bufio.Scanner`](https://pkg.go.dev/bufio#Scanner); though given that the list should be small and we're just moving code, not changing this is fine too.

Comment (adamkpickering) [reply src/go/rdctl/pkg/wsl/wsl_windows.go:40]: I think I will leave it as it is for now - I'd rather just get this code moved.

File: src/go/rdctl/pkg/factoryreset/factory_reset_windows.go (hunk 1 above)
Comment (adamkpickering) [reply src/go/rdctl/pkg/factoryreset/factory_reset_windows.go:38]: Whoops, I missed that. Fixed!

**Review by mook-as — APPROVED** [review]
I think it'd be good to have somebody more familiar with snapshots to do a quick pass too, but I'm generally fine with this.

**Review by jandubois — APPROVED** [review]
I have not tested the changes, but it makes sense to move the logic into the `wsl` package, and the PR just seems to move the implementation around.

**Pull request merged**